src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -26,163 +26,155 @@
-            }
-        } catch (Exception $e) {
-            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
-        }
+    /**
+     * @var string
+     */
+    public $php_self;
 
-        return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
-            'featureForm' => $featureForm->createView(),
-        ]);
-    }
+    /**
+     * @var ControllerConfiguration
+     */
+    public $controllerConfiguration;
 
     /**
-     * Edit feature action.
-     *
-     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
-     *
-     * @param int $featureId
-     * @param Request $request
-     *
-     * @return Response
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
      */
-    public function editAction($featureId, Request $request)
+    public function indexAction(Request $request)
     {
-        try {
-            $editableFeature = $this->getQueryBus()->handle(new GetFeatureForEditing((int) $featureId));
-        } catch (Exception $e) {
-            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
-
-            // @todo change route to features index when it's migrated
-            return $this->redirectToRoute('admin_features_create');
-        }
+        $this->setListFields();
+        $this->buildActionList();
+
+        $helperListConfiguration = $this->get('prestashop.core.bridge.helper_list_configuration_factory')->create([
+            'table' => self::TABLE,
+            'listId' => self::LIST_ID,
+            'className' => self::CLASS_NAME,
+            'controllerNameLegacy' => self::CONTROLLER_NAME_LEGACY,
+            'identifier' => self::IDENTIFIER,
+            'isJoinLanguageTableAuto' => true,
+            'defaultOrderBy' => 'position',
+            'fieldsList' => $this->controllerConfiguration->fieldsList,
+        ]);
 
-        if (!$this->isFeatureEnabled()) {
-            return $this->renderEditForm([
-                'showDisabledFeatureWarning' => true,
-                'editableFeature' => $editableFeature,
-            ]);
+        if ($request->request->has('submitResetfeature')) {
+            $this->get('prestashop.core.bridge.processor.reset_filter')->resetFilters($helperListConfiguration, $request);
         }
 
-        $featureFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.feature_form_builder');
-        $featureFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.feature_form_handler');
+        $this->get('prestashop.core.bridge.processor.process_filter')->processFilter(
+            $request,
+            $helperListConfiguration
+        );
 
-        $featureForm = $featureFormBuilder->getFormFor($featureId);
-        $featureForm->handleRequest($request);
+        $this->get('prestashop.core.bridge.helper_list_bridge')->getList(
```

Comment:
Why are `getList` and `renderList` separate actions? `getList` doesn't return anything, it just manipulates `$helperListConfiguration` before it's rendered.

Reply:
I think we might be adding too much complexity here. Consider this alternative:

```php
$list = new HelperList();
// set everything that needs to be set to HelperList, including defaults that used to be set by AdminController
// this allows us to avoid manipulating the helper's public properties
$listBridge->configure($list, $listConfiguration);

// we can modify the helper here if strictly necessary

// we now use the helper directly, and delegate the query and the structure to dedicated methods within the controller itself
$listContent = $list->generateList($this->getRows(), $this->getFields());

$this->renderSmarty($listContent, $this->controllerConfiguration);
```